packages/nimble-components/src/select/index.ts

```diff
@@ -722,6 +816,20 @@ export class Select
         }
     }
 
+    /**
+     * Fork of Listbox implementation to prevent placeholder option from being selected.
+     */
+    protected override getTypeaheadMatches(): ListboxOption[] {
```

Comment:
Instead of forking the entire implementation, what if you called the super class and then further filtered the result? Then, we wouldn't have to maintain the regex logic within nimble. Something like this:

```ts
protected override getTypeaheadMatches(): ListboxOption[] {
    const matchingOptions = super.getTypeaheadMatches();
    return matchingOptions.filter(o => !o.hidden & !o.disabled);
}
```

Technically, my suggestion is a little less performant because it's filtering the list twice, but since the second time only filters a subset of items (and we're talking a somewhat limited number of options), I'm not sure if that's a real concern.

Reply:
Done.